matching_test.go

```diff
@@ -5,27 +5,48 @@ import (
 	"testing"
 )
 
-var (
-	customError    = errors.New("custom error")
-	differentError = errors.New("different error")
+type simpleTestError struct{}
 
-	tests = map[error]error{
+func (c *simpleTestError) Error() string {
+	return "simpleTestError"
+}
+
+type wrapTestError struct {
+	err error
+}
+
+func (c *wrapTestError) Error() string {
+	return c.err.Error()
+}
+
+func (c *wrapTestError) Unwrap() error {
+	return c.err
+}
```

Comment:
I'd move all of that to `Test_ErrorMatching` as well.

Reply:
Unfortunately golang does not support nested function declaration. I can use `backoff_test` as a package to keep all test declarations from polluting library namespace.

Reply:
No let’s go with it. Those are private types anyway.